### PR TITLE
[v7r1]Fix discrepancy in CE _reset()

### PR DESCRIPTION
**Pull request by aldbr**
There is a discrepancy between the definition of the `_reset()` method in LocalCE and SSHCE, and in the others.
LocalCE and SSHCE return a value (`S_OK`/`S_ERROR`) while the others don't.

https://github.com/DIRACGrid/DIRAC/blob/integration/Resources/Computing/CREAMComputingElement.py#L102
https://github.com/DIRACGrid/DIRAC/blob/integration/Resources/Computing/LocalComputingElement.py#L71

Errors are not handled as ComputingElement ignores the value returned by _reset.

https://github.com/DIRACGrid/DIRAC/blob/integration/Resources/Computing/ComputingElement.py#L234

In this PR, I add a value to return in every definition of `_reset` and the value that it returns is considered.

BEGINRELEASENOTES
*Resources
FIX: discrepancy in CE._reset()
ENDRELEASENOTES

### CODE REVIEW

File: Resources/Computing/ComputingElement.py [hunk 1]
```diff
@@ -231,7 +231,9 @@ def setParameters(self, ceOptions):
       if key in FLOAT_PARAMETERS:
         self.ceParameters[key] = float(self.ceParameters[key])
 
-    self._reset()
+    result = self._reset()
+    if not result['OK']:
+      return result
     return S_OK()
```
Comment (fstagni) [inline Resources/Computing/ComputingElement.py]: The last 4 lines can simply be 
```
return self._reset()
```

File: Resources/Computing/ComputingElementFactory.py [hunk 2]
```diff
@@ -58,11 +58,13 @@ def getCE(self, ceType='', ceName='', ceParametersDict={}):
       ceDict = {'CEType': ceTypeLocal}
       if ceParametersDict:
         ceDict.update(ceParametersDict)
-      computingElement.setParameters(ceDict)
+      result = computingElement.setParameters(ceDict)
+      if not result['OK']:
+        return result
+
     except BaseException as x:
```
Comment (fstagni) [inline Resources/Computing/ComputingElementFactory.py]: Please replace `BaseException` with `Exception`

File: Resources/Computing/ComputingElementFactory.py [hunk 3]
```diff
@@ -58,11 +58,13 @@ def getCE(self, ceType='', ceName='', ceParametersDict={}):
...
-      msg = 'ComputingElementFactory could not instantiate %s object: %s' % (subClassName, str(x))
-      self.log.exception()
-      self.log.warn(msg)
+      msg = 'ComputingElementFactory could not instantiate %s object' % subClassName
+      self.log.exception(msg)
```
Comment (fstagni) [inline Resources/Computing/ComputingElementFactory.py]: Why? Honestly I prefer how it was written before.

File: Resources/Computing/LocalComputingElement.py [hunk 4]
```diff
@@ -68,6 +68,7 @@ def _reset(self):
 
     result = self._prepareHost()
     if not result['OK']:
+      self.log.error('Failed to initialize %s' % self.ceName)
```
Comment (fstagni) [inline Resources/Computing/LocalComputingElement.py]:
```suggestion
      self.log.error('Failed to initialize CE', self.ceName)
```

File: Resources/Computing/LocalComputingElement.py [hunk 5]
```diff
@@ -123,12 +124,12 @@ def _prepareHost(self):
     self.log.verbose('Creating working directories')
     result = systemCall(30, cmdTuple)
     if not result['OK']:
-      self.log.warn('Failed creating working directories: %s' % result['Message'][1])
+      self.log.error('Failed creating working directories: %s' % result['Message'][1])
```
Comment (fstagni) [inline Resources/Computing/LocalComputingElement.py]:
```suggestion
      self.log.error('Failed creating working directories', '(%s)' % result['Message'][1])
```

File: Resources/Computing/LocalComputingElement.py [hunk 6]
```diff
@@ -123,12 +124,12 @@ def _prepareHost(self):
...
       return result
-    status, output, _error = result['Value']
+    status, output, error = result['Value']
     if status != 0:
-      self.log.warn('Failed to create directories: %s' % output)
-      return S_ERROR('Failed to create directories: %s' % output)
+      self.log.error('Failed to create directories: %s' % error)
+      return S_ERROR('Failed to create directories: %s' % error)
```
Comment (fstagni) [inline Resources/Computing/LocalComputingElement.py]:
```suggestion
      return S_ERROR('Failed to create directories')
```

File: Resources/Computing/LocalComputingElement.py [hunk 7]
```diff
@@ -123,12 +124,12 @@ def _prepareHost(self):
...
     if not result['OK']:
-      self.log.warn('Failed creating working directories: %s' % result['Message'][1])
+      self.log.error('Failed creating working directories: %s' % result['Message'][1])
       return result
-    status, output, _error = result['Value']
+    status, output, error = result['Value']
     if status != 0:
-      self.log.warn('Failed to create directories: %s' % output)
-      return S_ERROR('Failed to create directories: %s' % output)
+      self.log.error('Failed to create directories: %s' % error)
```
Comment (fstagni) [inline Resources/Computing/LocalComputingElement.py]:
```suggestion
      self.log.error('Failed to create directories', '(%s)' % error)
```

File: Resources/Computing/SSHBatchComputingElement.py [hunk 8]
```diff
@@ -69,7 +69,8 @@ def _reset(self):
         self.log.info('Host %s registered for usage' % host)
         self.sshHost.append(hPar.strip())
       else:
-        self.log.error('Failed to initialize host', host)
+        self.log.error('Failed to initialize host %s' % host)
```
Comment (fstagni) [inline Resources/Computing/SSHBatchComputingElement.py]:
```suggestion
        self.log.error('Failed to initialize host', host)
```

File: Resources/Computing/SSHComputingElement.py [hunk 9]
```diff
@@ -324,6 +324,7 @@ def _reset(self):
 
     result = self._prepareRemoteHost()
     if not result['OK']:
+      self.log.error('Failed to initialize %s' % self.ceName)
```
Comment (fstagni) [inline Resources/Computing/SSHComputingElement.py]:
```suggestion
      self.log.error('Failed to initialize CE', self.ceName)
```

File: Resources/Computing/SSHComputingElement.py [hunk 10]
```diff
@@ -356,14 +357,14 @@ def _prepareRemoteHost(self, host=None):
     self.log.verbose('Creating working directories on %s' % self.ceParameters['SSHHost'])
     result = ssh.sshCall(30, cmd)
     if not result['OK']:
-      self.log.warn('Failed creating working directories: %s' % result['Message'][1])
+      self.log.error('Failed creating working directories: %s' % result['Message'][1])
```
Comment (fstagni) [inline Resources/Computing/SSHComputingElement.py]:
```suggestion
      self.log.error('Failed creating working directories', '(%s)' % result['Message'][1])
```

File: Resources/Computing/SSHComputingElement.py [hunk 11]
```diff
@@ -356,14 +357,14 @@ def _prepareRemoteHost(self, host=None):
...
       return result
     status, output, _error = result['Value']
     if status == -1:
-      self.log.warn('Timeout while creating directories')
+      self.log.error('Timeout while creating directories')
       return S_ERROR('Timeout while creating directories')
     if "cannot" in output:
-      self.log.warn('Failed to create directories: %s' % output)
+      self.log.error('Failed to create directories: %s' % output)
```
Comment (fstagni) [inline Resources/Computing/SSHComputingElement.py]:
```suggestion
      self.log.error('Failed to create directories', '(%s)' % output)
```

And for this one, and other "standard", please use `errno` or `DErrno` codes.

File: Resources/Computing/ComputingElementFactory.py (hunk 3 above)
Comment (aldbr) [reply Resources/Computing/ComputingElementFactory.py]: I thought the `warn` was unnecessary. Why do you prefer it like this?
I canceled my change.

**Pull request merged**